src/crud.service.ts

```diff
@@ -40,128 +40,128 @@ export class CrudService<T> extends CommonService
 	/**
 	 * fires a request to create a single resource
 	 *
-	 * @since 1.0.0
+	 * @since 8.0.0
 	 *
 	 * @param {BodyInterface} body body of the request
 	 * @param {OptionInterface} options options of the request
 	 *
-	 * @return {Observable<T>} http response
+	 * @return {Observable<$>} http response
 	 */
 
-	public create(body : BodyInterface, options ?: OptionInterface) : Observable<T>
+	public create<$ = T>(body : BodyInterface, options ?: OptionInterface) : Observable<$>
 	{
-		return this.postService.bind(this).post(body, options);
+		return this.postService.bind(this).post<$>(body, options);
 	}
 
 	/**
 	 * fires a request to read a single resource
 	 *
-	 * @since 1.0.0
+	 * @since 8.0.0
 	 *
 	 * @param {IdType} id identifier of the resource
 	 * @param {OptionInterface} options options of the request
 	 *
-	 * @return {Observable<T>} http response
+	 * @return {Observable<$>} http response
 	 */
 
-	public read(id : IdType, options ?: OptionInterface) : Observable<T>
+	public read<$ = T>(id : IdType, options ?: OptionInterface) : Observable<$>
 	{
-		return this.getService.bind(this).get(id, options);
+		return this.getService.bind(this).get<$>(id, options);
 	}
 
 	/**
 	 * fires a request to find multiple resources
 	 *
-	 * @since 1.0.0
+	 * @since 8.0.0
 	 *
 	 * @param {OptionInterface} options options of the request
 	 *
-	 * @return {Observable<T[]>} http response
+	 * @return {Observable<$>} http response
 	 */
 
-	public find(options ?: OptionInterface) : Observable<T[]>
+	public find<$ = T[]>(options ?: OptionInterface) : Observable<$>
```

Comment:
Can you elaborate a bit more what problem you experience here?

I read the following in the description

```

This should be invalid but return type overrides find() typing:

public findByUser(userId : string) : Observable<TestInterface>
{
	return this.setParam('userId', userId).find();
}
```

What do you expectation?
Does find return `TestInterface[]` but the compiler does not explain as expected?

In other words, what return type overrides `find`?

Reply:
The compiler does not shouw me an error as `Observable<TestInterface>` returned by `findByUser()` does not match `Observable<TestInterface[]>` by `find()` ... that being said, it silenty overrides it.